pkg/api/conversions/conversions.go

```diff
@@ -0,0 +1,103 @@
+package migrate
+
+import (
+	"encoding/json"
+	"fmt"
+
+	machinev1alpha1downstream "github.com/kubermatic/machine-controller/pkg/machines/v1alpha1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	runtime "k8s.io/apimachinery/pkg/runtime"
+	clustercommon "sigs.k8s.io/cluster-api/pkg/apis/cluster/common"
+	clusterv1alpha1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+)
+
+func ConvertV1alpha1DownStreamMachineToV1alpha1ClusterMachine(in machinev1alpha1downstream.Machine) (*clusterv1alpha1.Machine, error) {
+	out := &clusterv1alpha1.Machine{}
+	out.ObjectMeta = in.ObjectMeta
+	out.Spec.ObjectMeta = in.Spec.ObjectMeta
+	out.SelfLink = ""
```

Comment:
will `out.APIVersion` and `Kind` be assigned by the API server ?

Reply:
I assumed this will implicitly be there because it is a `clusterv1alpha1.Machine{}`